diff --git a/__init__.py b/__init__.py
index 542ecda..4b63cc6 100644
--- a/__init__.py
+++ b/__init__.py
@@ -149,6 +149,10 @@ class WolframAlphaSkill(MycroftSkill):
 
             # TODO: Log missed intent
             LOG.debug("Unknown intent: " + utterance)
+            self.emitter.emit(Message("fallback_failure", {
+                "utterance": utt,
+                "lang": lang
+                }))
             return
 
         try:
