diff --git a/mycroft/tts/__init__.py b/mycroft/tts/__init__.py
index 88ff97e..057978a 100644
--- a/mycroft/tts/__init__.py
+++ b/mycroft/tts/__init__.py
@@ -385,6 +385,7 @@ class TTSFactory(object):
     from mycroft.tts.spdsay_tts import SpdSay
     from mycroft.tts.bing_tts import BingTTS
     from mycroft.tts.ibm_tts import WatsonTTS
+    from mycroft.tts.arlobot_tts import ArlobotTTS
 
     CLASSES = {
         "mimic": Mimic,
@@ -394,7 +395,8 @@ class TTSFactory(object):
         "espeak": ESpeak,
         "spdsay": SpdSay,
         "watson": WatsonTTS,
-        "bing": BingTTS
+        "bing": BingTTS,
+        "arlobot": ArlobotTTS
     }
 
     @staticmethod
